content/blog/2017/08/2017-08-10-kubernetes-with-pipeline-acs.adoc

```diff
@@ -0,0 +1,156 @@
+link:https://jenkinsworld20162017.sched.com/event/AK4K/azure-devops-open-source-integrations[Azure
+DevOps open source integrations], in this blog post I wanted to share some good
+snippets of what is possible today with Jenkins Pipeline and Azure.
+
+First, a simple example using the Azure CLI to list resources in the
+subscription:
+
+[pipeline]
+----
+// Scripted //
+node {
+    /* .. snip .. */
+    stage('Deploy') {
+        withCredentials([azureServicePrincipal('principal-credentials-id')]) {
+            sh 'az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID'
+            sh 'az account set -s $AZURE_SUBSCRIPTION_ID'
+            sh 'az resource list'
+        }
+    }
+}
+----
+
+[WARNING]
+====
+`azureServicePrincipal()` cannot be used in Declarative Pipeline until
+link:https://issues.jenkins-ci.org/browse/JENKINS-46103[JENKINS-46103] is
+resolved.
+====
+
+Once a Pipeline can interact with Azure, there are countless ways one could
+implement continuous delivery with Jenkins and Azure. From a deploying a simple
+webapp with the
+link:https://azure.microsoft.com/en-us/blog/annoucing-jenkins-deploy-to-azure-app-service-plugin-and-new-managed-disk-support-for-azure-storage-plugin/[Azure
+App Service plugin] and the `azureWebAppPublish` step, or a more advanced
+container-based delivery pipeline to deliver new containers to
+link:https://docs.microsoft.com/en-us/azure/container-service/kubernetes/container-service-tutorial-kubernetes-deploy-cluster[Kubernetes
+via Azure Container Service].
+
+With the plugin:docker-workflow[Docker Pipeline plugin] and a little bit of
+extra scripting, a Jenkins Pipeline can also build and publish a Docker
+container to an
+link:https://azure.microsoft.com/en-us/services/container-registry/[Azure
+Container Registry]:
+
+[pipeline]
+----
+// Scripted //
+import groovy.json.JsonSlurper
+
+node {
+    def container
+    def acrSettings
+
+    withCredentials([azureServicePrincipal('principal-credentials-id')]) {
+        stage('Prepare Environment') {
+            sh 'az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID'
+            sh 'az account set -s $AZURE_SUBSCRIPTION_ID'
+            acrSettings = new JsonSlurper().parseText(
+                                            sh(script: "az acs show -o json -n my-acr", returnStdout: true))
+        }
+
+        stage('Build') {
+            container = docker.build("${acrSettings.loginServer}/my-app:${env.BUILD_ID}")
+        }
+
+        stage('Publish') {
+            /* https://issues.jenkins-ci.org/browse/JENKINS-46108 */
+            sh "docker login -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} ${acrSettings.loginServer}"
+            container.push()
+        }
+
+        stage('Deploy') {
+            echo 'Orchestrating a new deployment with kubectl is a simple exercise left to the reader ;)'
+        }
+    }
+}
+----
+
+---
```

Comment:
That's a horizontal rule in asciidoc which I put t here intentionally :smile_cat: